lib/src/cli/flutter_cli.dart

```diff
@@ -367,41 +368,52 @@ Future<int> _flutterTest({
         if (event.stackTrace.trim().isNotEmpty) {
           stderr('$clearLine${event.stackTrace}');
         }
-
-        final traceLocation = _getTraceLocation(stackTrace: event.stackTrace);
-
-        // When failing to recover the location from the stack trace,
-        // save a short description of the error
-        final testErrorDescription = traceLocation ??
-            event.error.replaceAll('\n', ' ').truncated(_lineLength);
-
+        final test = tests[event.testID]!;
+        final suite = suites[test.suiteID]!;
         final prefix = event.isFailure ? '[FAILED]' : '[ERROR]';
-        failedTestErrorMessages[event.testID] = '$prefix $testErrorDescription';
+
+        final optimizationApplied = _isOptimizationApplied(suite);
+        final firstGroupName = _firstGroupName(test, groups);
```

Comment:
`_firstGroupName` can be better described as "top group"

Reply:
Var renamed to topGroupeName. Done.